packages/types/src/plugin.ts

```diff
@@ -11,11 +11,14 @@ import {
   DefaultArgs,
 } from './hooks';
 
-export interface Plugin<PluginContext extends Record<string, any> = {}> {
+export interface Plugin<
+  TInputContext extends Record<string, any> = {},
+  TOutputContext extends Record<string, any> = TInputContext
```

Comment:
Does it work?
```suggestion
  TOutputContext extends TInputContext
```

If so it would be nice to say that you need to return something that at least extends what you were given, without removing anything.

Reply:
Theoretically, the plugin can overwrite a property in the existing context by calling `extendContext({ property: null })`. So extending might be wrong here.

Reply:
Maybe it is something you want to allow? Otherwise it might be nice to have type checks prohibiting accidentally ruining the context chain. 

Wrong:
```
return { property: 'foo' } // type error: { property: 'foo' } is not assignable to TInputContext
```

Right:
```
return { ...inputContext, property: 'foo' }
```